subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppLibraryIntegrationTest.groovy

```diff
@@ -152,6 +152,31 @@ class CppLibraryIntegrationTest extends AbstractCppInstalledToolChainIntegration
         objectFiles(lib.sources)*.assertExists()
     }
 
+    def "can build static and shared variants of library"() {
+        given:
+        settingsFile << "rootProject.name = 'hello'"
+        def lib = new CppGreeterWithOptionalFeature()
+        lib.writeToProject(testDirectory)
+
+        and:
+        buildFile << """
+            apply plugin: 'cpp-library'
+         """
+
+        expect:
+        succeeds "linkDebugShared"
+
+        result.assertTasksExecuted(compileAndLinkTasks(debug))
+        sharedLibrary("build/lib/main/debug/shared/hello").assertExists()
+        staticLibrary("build/lib/main/debug/static/hello").assertDoesNotExist()
```

Comment:
we need to be careful with 'does not exist' assertions in tests, these are unreliable unless we also verify 'exists' of the same thing somewhere else in the test.

Reply:
Done.